external-crates/move/crates/move-bytecode-verifier/src/reference_safety/abstract_state.rs

```diff
@@ -452,6 +473,45 @@ impl AbstractState {
         Ok(AbstractValue::Reference(field_borrow_id))
     }
 
+    pub fn unpack_enum_variant_ref(
+        &mut self,
+        offset: CodeOffset,
+        variant_tag: VariantTag,
+        variant_def: &VariantDefinition,
+        mut_: bool,
+        id: RefID,
+    ) -> PartialVMResult<Vec<AbstractValue>> {
+        // Any field borrows will be factored out, so don't check in the mutable case
+        let is_mut_borrow_with_full_borrows = || mut_ && self.has_full_borrows(id);
```

Comment:
Have I mentioned how much I have been wanting to rewrite this fuckin thing?... sigh... someday